# Information Retrieval Project (phase 2)
### By Ali Moradzade 9831058


In [1]:
from datetime import date

today = date.today()
print("Today's date:", today)

Today's date: 2022-06-08


# Importing some useful libraries 

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Reading the data

For real purposes we use uncomment the below line:

In [3]:
df = pd.read_json('data/IR_data_news_12k.json')

For testing the functionality we use just first 10 entries: (for real use, comment it)

In [4]:
# df = df.transpose().iloc[0:10].transpose()

## Doing some EDA

In [5]:
df.head()

0      \
title       اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا   
content  \nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آس...   
tags     [اعلام زمان, قرعه‌کشی, قرعه‌کشی جام, قرعه‌کشی ...   
date                                  3/15/2022 5:59:27 PM   
url      https://www.farsnews.ir/news/14001224001005/اع...   

                                                     1      \
title    سجادی :حضور تماشاگران در  لیگ برتر فوتبال تابع...   
content  \nبه گزارش خبرگزاری فارس، سید حمید سجادی در حا...   
tags                     [سجادی, لیگ, فدراسیون, وزیر ورزش]   
date                                  3/15/2022 5:30:07 PM   
url      https://www.farsnews.ir/news/14001224000982/سج...   

                                                     2      \
title    محل برگزاری نشست‌های خبری سرخابی‌ها؛ مجیدی در ...   
content  \nبه گزارش خبرگزاری فارس، نشست خبری پیش از مسا...   
tags     [دربی 94, محل برگزاری, خبری سرخابی‌ها, مجیدی, ...   
date                                  3/15/2022 5:20:01 PM   
url      https://www.farsnews.ir/news/14001224000971/مح...   

                                                     3      \
title    ماجدی در نشست با صالحی امیری: امیدوارم در این ...   
content  \nبه گزارش خبرگزاری فارس،  سید رضا صالحی امیری...   
tags     [کمیته امداد امام خمینی (ره), کمیته ملی المپیک...   
date                                  3/15/2022 5:18:00 PM   
url      https://www.farsnews.ir/news/14001224000964/ما...   

                                                     4      \
title    لیگ‌برتر بسکتبال|‌ نخستین پیروزی شهرداری گرگان...   
content  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، در نخس...   
tags     [بسکتبال, لیگ برتر بسکتبال, شهرداری گرگان, تیم...   
date                                  3/15/2022 5:16:41 PM   
url      https://www.farsnews.ir/news/14001224000947/لی...   

                                                     5      \
title    مسابقات تنیس روی میز فیدر قطر| هر4 بانوی ملی پ...   
content  \nبه گزارش خبرگزاری فارس، چهار بانوی تنیس روی ...   
tags     [مهشید اشتری, تنیس روی میز, ملی پوشان تنیس روی...   
date                                  3/15/2022 5:15:34 PM   
url      https://www.farsnews.ir/news/14001224000916/مس...   

                                                     6      \
title    اعلام برنامه نشست خبری گل محمدی/ مجیدی هم باید...   
content  \nبه گزارش خبرگزاری فارس و به نقل از  سایت باش...   
tags     [باشگاه استقلال, یحیی گل محمدی, فرهاد مجیدی, ن...   
date                                  3/15/2022 3:57:47 PM   
url      https://www.farsnews.ir/news/14001224000865/اع...   

                                                     7      \
title    احضار مدیران پرسپولیس به کمیته انضباطی  پیش از...   
content  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، کمیته ...   
tags     [باشگاه پرسپولیس, احضار مدیران پرسپولیس به کمی...   
date                                  3/15/2022 3:43:18 PM   
url      https://www.farsnews.ir/news/14001224000842/اح...   

                                                     8      \
title    مدیر بی‌دستاورد، رئیس شد/ روزگار تلخ‌تر از تلخ...   
content  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، مجمع ا...   
tags     [والیبال, پیمان رضایی, تیم والیبال شهرداری ارو...   
date                                  3/15/2022 3:35:21 PM   
url      https://www.farsnews.ir/news/14001224000811/مد...   

                                                     9      ...  \
title        خبر خوب برای استقلال؛ دانشگر با تیم تمرین کرد  ...   
content  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، تمرین ...  ...   
tags       [دانشگر با تیم تمرین کرد, محمد دانشگر, استقلال]  ...   
date                                  3/15/2022 3:21:04 PM  ...   
url      https://www.farsnews.ir/news/14001224000819/خب...  ...   

                                                     12192  \
title    کمیته ویژه‌ای از سوی رئیس جمهور مسئول پیگیری م...   
content  \nاسماعیلی حسین زهی نماینده مردم خاش در مجلس ش...   
tags     [مجلس شورای اسلامی, کمیسیون عمران, اسماعیل حسی...   
date                                 10/17/20

In [6]:
df = df.transpose()

In [7]:
df.head()

title  \
0     اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا   
1  سجادی :حضور تماشاگران در  لیگ برتر فوتبال تابع...   
2  محل برگزاری نشست‌های خبری سرخابی‌ها؛ مجیدی در ...   
3  ماجدی در نشست با صالحی امیری: امیدوارم در این ...   
4  لیگ‌برتر بسکتبال|‌ نخستین پیروزی شهرداری گرگان...   

                                             content  \
0  \nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آس...   
1  \nبه گزارش خبرگزاری فارس، سید حمید سجادی در حا...   
2  \nبه گزارش خبرگزاری فارس، نشست خبری پیش از مسا...   
3  \nبه گزارش خبرگزاری فارس،  سید رضا صالحی امیری...   
4  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، در نخس...   

                                                tags                  date  \
0  [اعلام زمان, قرعه‌کشی, قرعه‌کشی جام, قرعه‌کشی ...  3/15/2022 5:59:27 PM   
1                  [سجادی, لیگ, فدراسیون, وزیر ورزش]  3/15/2022 5:30:07 PM   
2  [دربی 94, محل برگزاری, خبری سرخابی‌ها, مجیدی, ...  3/15/2022 5:20:01 PM   
3  [کمیته امداد امام خمینی (ره), کمیته ملی المپیک...  3/15/2022 5:18:00 PM   
4  [بسکتبال, لیگ برتر بسکتبال, شهرداری گرگان, تیم...  3/15/2022 5:16:41 PM   

                                                 url category  
0  https://www.farsnews.ir/news/14001224001005/اع...   sports  
1  https://www.farsnews.ir/news/14001224000982/سج...   sports  
2  https://www.farsnews.ir/news/14001224000971/مح...   sports  
3  https://www.farsnews.ir/news/14001224000964/ما...   sports  
4  https://www.farsnews.ir/news/14001224000947/لی...   sports

In [8]:
df = df.reset_index().rename(columns={'index': 'id'})

In [9]:
df.head()

id                                              title  \
0   0     اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا   
1   1  سجادی :حضور تماشاگران در  لیگ برتر فوتبال تابع...   
2   2  محل برگزاری نشست‌های خبری سرخابی‌ها؛ مجیدی در ...   
3   3  ماجدی در نشست با صالحی امیری: امیدوارم در این ...   
4   4  لیگ‌برتر بسکتبال|‌ نخستین پیروزی شهرداری گرگان...   

                                             content  \
0  \nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آس...   
1  \nبه گزارش خبرگزاری فارس، سید حمید سجادی در حا...   
2  \nبه گزارش خبرگزاری فارس، نشست خبری پیش از مسا...   
3  \nبه گزارش خبرگزاری فارس،  سید رضا صالحی امیری...   
4  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، در نخس...   

                                                tags                  date  \
0  [اعلام زمان, قرعه‌کشی, قرعه‌کشی جام, قرعه‌کشی ...  3/15/2022 5:59:27 PM   
1                  [سجادی, لیگ, فدراسیون, وزیر ورزش]  3/15/2022 5:30:07 PM   
2  [دربی 94, محل برگزاری, خبری سرخابی‌ها, مجیدی, ...  3/15/2022 5:20:01 PM   
3  [کمیته امداد امام خمینی (ره), کمیته ملی المپیک...  3/15/2022 5:18:00 PM   
4  [بسکتبال, لیگ برتر بسکتبال, شهرداری گرگان, تیم...  3/15/2022 5:16:41 PM   

                                                 url category  
0  https://www.farsnews.ir/news/14001224001005/اع...   sports  
1  https://www.farsnews.ir/news/14001224000982/سج...   sports  
2  https://www.farsnews.ir/news/14001224000971/مح...   sports  
3  https://www.farsnews.ir/news/14001224000964/ما...   sports  
4  https://www.farsnews.ir/news/14001224000947/لی...   sports

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12202 entries, 0 to 12201
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        12202 non-null  int64 
 1   title     12202 non-null  object
 2   content   12202 non-null  object
 3   tags      12202 non-null  object
 4   date      12202 non-null  object
 5   url       12202 non-null  object
 6   category  12202 non-null  object
dtypes: int64(1), object(6)
memory usage: 667.4+ KB


In [11]:
df.describe().transpose()

count    mean          std  min      25%     50%      75%      max
id  12202.0  6100.5  3522.558327  0.0  3050.25  6100.5  9150.75  12201.0

In [12]:
df.iloc[0]

id                                                          0
title          اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا
content     \nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آس...
tags        [اعلام زمان, قرعه‌کشی, قرعه‌کشی جام, قرعه‌کشی ...
date                                     3/15/2022 5:59:27 PM
url         https://www.farsnews.ir/news/14001224001005/اع...
category                                               sports
Name: 0, dtype: object

In [13]:
df.iloc[0]['title']

'اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا'

In [14]:
df.iloc[0]['content']

'\nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان\xa0 قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/\n\n\n'

In [15]:
print(df.iloc[0]['content'])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [16]:
df.iloc[0]['tags']

['اعلام زمان',
 'قرعه\u200cکشی',
 'قرعه\u200cکشی جام',
 'قرعه\u200cکشی جام باشگاه\u200cهای فوتسال',
 'ای اف سی',
 'گیتی پسند']

In [17]:
for item in df.iloc[0]['tags']:
    print(item)

اعلام زمان
قرعه‌کشی
قرعه‌کشی جام
قرعه‌کشی جام باشگاه‌های فوتسال
ای اف سی
گیتی پسند


In [18]:
df.iloc[0]['date']

'3/15/2022 5:59:27 PM'

In [19]:
df.iloc[0]['url']

'https://www.farsnews.ir/news/14001224001005/اعلام-زمان-قرعه-کشی-جام-باشگاه-های-فوتسال-آسیا'

# Preprocessing

## Deleting some extra stuff in content

In [20]:
import re

def del_end_of_message(message):
    return message.replace('انتهای پیام/', '')

In [21]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [22]:
print(del_end_of_message(df['content'].iloc[0]))


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. 





## Deleting punctuation 

In [23]:
import string

def del_punc(content):
    puncs = string.punctuation + "".join(['؛', '_', '،', '؟', '؛', '»', '«', 'ـ', '،'])
    
    return content.translate(content.maketrans("", "", puncs))

In [24]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [25]:
print(del_punc(df['content'].iloc[0]))


به گزارش خبرگزاری فارس کنفدراسیون فوتبال آسیا AFC در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است انتهای پیام





## Doing normalization

In [26]:
from parsivar import Normalizer

normalizer = Normalizer(statistical_space_correction=True)
def normalize(content):
    return normalizer.normalize(content)

In [27]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [28]:
print(normalize(df['content'].iloc[0]))

به گزارش خبرگزاری فارس ، کنفدراسیون فوتبال آسیا ( AFC ) در نامه‌ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان قرعه‌کشی جام باشگاه‌های فوتسال آسیا را رسما اعلام کرد . بر این اساس 25 فروردین ماه 1401 مراسم قرعه‌کشی جام باشگاه‌های فوتسال آسیا در مالزی برگزار می‌شود . باشگاه گیتی‌پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است . پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه‌های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست‌آورده است . انتهای پیام / 


## Doing tokenization

In [29]:
from hazm import word_tokenize

In [30]:
def tokenize(content):
    return word_tokenize(content)

In [31]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [32]:
tokenize(df['content'].iloc[0])

['به',
 'گزارش',
 'خبرگزاری',
 'فارس',
 '،',
 'کنفدراسیون',
 'فوتبال',
 'آسیا',
 '(',
 'AFC',
 ')',
 'در',
 'نامه',
 'ای',
 'رسمی',
 'به',
 'فدراسیون',
 'فوتبال',
 'ایران',
 'و',
 'باشگاه',
 'گیتی',
 'پسند',
 'زمان\xa0',
 'قرعه',
 'کشی',
 'جام',
 'باشگاه',
 'های',
 'فوتسال',
 'آسیا',
 'را',
 'رسماً',
 'اعلام',
 'کرد',
 '.',
 'بر',
 'این',
 'اساس',
 '25',
 'فروردین',
 'ماه',
 '1401',
 'مراسم',
 'قرعه',
 'کشی',
 'جام',
 'باشگاه',
 'های',
 'فوتسال',
 'آسیا',
 'در',
 'مالزی',
 'برگزار',
 'می',
 'شود',
 '.',
 'باشگاه',
 'گیتی',
 'پسند',
 'بعنوان',
 'قهرمان',
 'فوتسال',
 'ایران',
 'در',
 'سال',
 '1400',
 'به',
 'این',
 'مسابقات',
 'راه',
 'پیدا',
 'کرده_است',
 '.',
 'پیش',
 'از',
 'این',
 'گیتی',
 'پسند',
 'تجربه',
 '3',
 'دوره',
 'حضور',
 'در',
 'جام',
 'باشگاه',
 'های',
 'فوتسال',
 'آسیا',
 'را',
 'داشته',
 'که',
 'هر',
 'سه',
 'دوره',
 'به',
 'فینال',
 'مسابقات',
 'راه',
 'پیدا',
 'کرده',
 'و',
 'یک',
 'عنوان',
 'قهرمانی',
 'و',
 'دو',
 'مقام',
 'دومی',
 'بدست',
 'آورده_است',
 '.',
 'انته

In [33]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





## Doing stemming

In [34]:
from parsivar import FindStems

stemmer = FindStems()

def stem(content):
    return stemmer.convert_to_stem(content)

In [35]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [36]:
for term in tokenize(df['content'].iloc[0]):
    print(stem(term), end=' ')

به گزارش خبرگزاری فارس ، کنفدراسیون فوتبال آسیا ( AFC ) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشید&کش جام باشگاه های فوتسال آسیا را رسماً اعلام کرد . بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشید&کش جام باشگاه های فوتسال آسیا در مالزی برگزار می شد&شو . باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده_است . پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده_است . انتهای پیام/ 

## Stop words

In [37]:
from hazm import stopwords_list

persian_stop_words = stopwords_list()

In [38]:
persian_stop_words

['و',
 'در',
 'به',
 'از',
 'که',
 'این',
 'را',
 'با',
 'است',
 'برای',
 'آن',
 'یک',
 'خود',
 'تا',
 'کرد',
 'بر',
 'هم',
 'نیز',
 'گفت',
 'می\u200cشود',
 'وی',
 'شد',
 'دارد',
 'ما',
 'اما',
 'یا',
 'شده',
 'باید',
 'هر',
 'آنها',
 'بود',
 'او',
 'دیگر',
 'دو',
 'مورد',
 'می\u200cکند',
 'شود',
 'کند',
 'وجود',
 'بین',
 'پیش',
 'شده_است',
 'پس',
 'نظر',
 'اگر',
 'همه',
 'یکی',
 'حال',
 'هستند',
 'من',
 'کنند',
 'نیست',
 'باشد',
 'چه',
 'بی',
 'می',
 'بخش',
 'می\u200cکنند',
 'همین',
 'افزود',
 'هایی',
 'دارند',
 'راه',
 'همچنین',
 'روی',
 'داد',
 'بیشتر',
 'بسیار',
 'سه',
 'داشت',
 'چند',
 'سوی',
 'تنها',
 'هیچ',
 'میان',
 'اینکه',
 'شدن',
 'بعد',
 'جدید',
 'ولی',
 'حتی',
 'کردن',
 'برخی',
 'کردند',
 'می\u200cدهد',
 'اول',
 'نه',
 'کرده_است',
 'نسبت',
 'بیش',
 'شما',
 'چنین',
 'طور',
 'افراد',
 'تمام',
 'درباره',
 'بار',
 'بسیاری',
 'می\u200cتواند',
 'کرده',
 'چون',
 'ندارد',
 'دوم',
 'بزرگ',
 'طی',
 'حدود',
 'همان',
 'بدون',
 'البته',
 'آنان',
 'می\u200cگوید',
 'دیگری',
 'خواهد_شد',


## Applying all of our methods:

In [39]:
from collections import Counter

def preprocess(content):
    
    # deleting end of message 
    content = del_end_of_message(content)
    
    # normalizing our content
    content = normalize(content)
    
    # removing punctuation
    content = del_punc(content)
    
    # doing both stemming and tokenization
    words = list(map(lambda term : stem(term), tokenize(content)))
    word_frequency = dict(Counter(words))
    
    # removing stopwords
    for stop_word in stopwords_list():
        if stop_word in word_frequency.keys():
            word_frequency.pop(stop_word)
            
    return word_frequency

In [40]:
print(df['content'].iloc[0])


به گزارش خبرگزاری فارس، کنفدراسیون فوتبال آسیا (AFC) در نامه ای رسمی به فدراسیون فوتبال ایران و باشگاه گیتی پسند زمان  قرعه کشی جام باشگاه های فوتسال آسیا را رسماً اعلام کرد. بر این اساس 25 فروردین ماه 1401 مراسم قرعه کشی جام باشگاه های فوتسال آسیا در مالزی برگزار می شود. باشگاه گیتی پسند بعنوان قهرمان فوتسال ایران در سال 1400 به این مسابقات راه پیدا کرده است. پیش از این گیتی پسند تجربه 3 دوره حضور در جام باشگاه های فوتسال آسیا را داشته که هر سه دوره به فینال مسابقات راه پیدا کرده و یک عنوان قهرمانی و دو مقام دومی بدست آورده است. انتهای پیام/





In [41]:
print(preprocess(df['content'].iloc[0]))

{'گزارش': 1, 'خبرگزاری': 1, 'فارس': 1, 'کنفدراسیون': 1, 'فوتبال': 2, 'آسیا': 4, 'AFC': 1, 'نامه': 1, 'رسمی': 1, 'فدراسیون': 1, 'ایران': 2, 'باشگاه': 5, 'گیتی': 2, 'پسند': 2, 'زمان': 1, 'قرعه\u200cکشی': 2, 'جام': 3, 'فوتسال': 4, 'رسما': 1, 'اعلام': 1, 'اساس': 1, '25': 1, 'فروردین': 1, 'ماه': 1, '1401': 1, 'مراسم': 1, 'مالزی': 1, 'برگزار': 1, 'شد&شو': 1, 'گیتی\u200cپسند': 1, 'بعنوان': 1, 'قهرمان': 1, 'سال': 1, '1400': 1, 'مسابقات': 2, 'تجربه': 1, '3': 1, 'دوره': 2, 'حضور': 1, 'فینال': 1, 'عنوان': 1, 'قهرمانی': 1, 'مقام': 1, 'دومی': 1, 'بدست\u200cآورده': 1, 'اس': 1}


---

Running this cell below will take:

    CPU times: user 4min 11s, sys: 346 ms, total: 4min 11s
    Wall time: 4min 11s

In [42]:
# %%time
# df['preprocessed'] = df['content'].apply(preprocess)

In [43]:
# df.head()

## Saving our preprocessed data frame 

In [44]:
# df.to_pickle('results/preprocessed-phase02.pickle')

## Loading our preprocessed result

In [45]:
ready_df = pd.read_pickle('results/preprocessed-phase02.pickle')

In [46]:
ready_df.head()

id                                              title  \
0   0     اعلام زمان قرعه کشی جام باشگاه های فوتسال آسیا   
1   1  سجادی :حضور تماشاگران در  لیگ برتر فوتبال تابع...   
2   2  محل برگزاری نشست‌های خبری سرخابی‌ها؛ مجیدی در ...   
3   3  ماجدی در نشست با صالحی امیری: امیدوارم در این ...   
4   4  لیگ‌برتر بسکتبال|‌ نخستین پیروزی شهرداری گرگان...   

                                             content  \
0  \nبه گزارش خبرگزاری فارس، کنفدراسیون فوتبال آس...   
1  \nبه گزارش خبرگزاری فارس، سید حمید سجادی در حا...   
2  \nبه گزارش خبرگزاری فارس، نشست خبری پیش از مسا...   
3  \nبه گزارش خبرگزاری فارس،  سید رضا صالحی امیری...   
4  \nبه گزارش خبرنگار ورزشی خبرگزاری فارس، در نخس...   

                                                tags                  date  \
0  [اعلام زمان, قرعه‌کشی, قرعه‌کشی جام, قرعه‌کشی ...  3/15/2022 5:59:27 PM   
1                  [سجادی, لیگ, فدراسیون, وزیر ورزش]  3/15/2022 5:30:07 PM   
2  [دربی 94, محل برگزاری, خبری سرخابی‌ها, مجیدی, ...  3/15/2022 5:20:01 PM   
3  [کمیته امداد امام خمینی (ره), کمیته ملی المپیک...  3/15/2022 5:18:00 PM   
4  [بسکتبال, لیگ برتر بسکتبال, شهرداری گرگان, تیم...  3/15/2022 5:16:41 PM   

                                                 url category  \
0  https://www.farsnews.ir/news/14001224001005/اع...   sports   
1  https://www.farsnews.ir/news/14001224000982/سج...   sports   
2  https://www.farsnews.ir/news/14001224000971/مح...   sports   
3  https://www.farsnews.ir/news/14001224000964/ما...   sports   
4  https://www.farsnews.ir/news/14001224000947/لی...   sports   

                                        preprocessed  
0  {'گزارش': 1, 'خبرگزاری': 1, 'فارس': 1, 'کنفدرا...  
1  {'گزارش': 1, 'خبرگزاری': 1, 'فارس': 1, 'سید': ...  
2  {'گزارش': 1, 'خبرگزاری': 1, 'فارس': 1, 'خبری':...  
3  {'گزارش': 1, 'خبرگزاری': 1, 'فارس': 1, 'سید': ...  
4  {'گزارش': 1, 'خبرنگار': 1, 'ورزشی': 1, 'خبرگزا...

In [47]:
df = ready_df

# Making dictionary and processing user queries

## Log-frequency weighting

In [48]:
def log_frequency_weighting(term_frequency):
    if term_frequency > 0:
        return 1 + math.log10(term_frequency)
    else:
        return 0

## Making dictionary

This below algorithm makes our dictionary.  
  
Our dictionary structure is as follows:

```
dictionary { term1 : [total number of occurances of term1 in collection, {doc id: the log frequency weight of term1}],
             term2 : ..
             ...
             termn: ..}
```

In [49]:
import math

def make_dictionary(data_frame):
    my_dictionary = {}
    doc_normalized_weights = {}
    
    for doc_id, doc_content in zip(data_frame['id'], data_frame['content']):
        word_frequencies_in_doc = preprocess(doc_content)
        total_doc_weight_squared = 0
        
        for word, word_frequency in word_frequencies_in_doc.items():
            total_doc_weight_squared += log_frequency_weighting(word_frequency) ** 2
            
            if word in my_dictionary.keys():
                my_dictionary[word][0] += 1
                my_dictionary[word][1][doc_id] = log_frequency_weighting(word_frequency)
                
            else:
                my_dictionary[word] = [1, {doc_id: log_frequency_weighting(word_frequency)}]
                
        doc_normalized_weights[doc_id] = np.sqrt(total_doc_weight_squared)
        
    return my_dictionary, doc_normalized_weights

This cell below will take:

    CPU times: user 4min 4s, sys: 240 ms, total: 4min 4s
    Wall time: 4min 4s

In [50]:
# %%time
# my_dict, lengths = make_dictionary(df)

## Save and load 

In [51]:
import pickle

### Saving our index

In [52]:
# import pickle

# with open('results/dictionary-phase02.pickle', 'wb') as file:
#     pickle.dump(my_dict, file, pickle.HIGHEST_PROTOCOL)

### Loading our index

In [53]:
with open('results/dictionary-phase02.pickle', 'rb') as file:
    loaded_dict = pickle.load(file)

In [54]:
my_dict = loaded_dict

### Saving our lengthes

In [55]:
# with open('results/lengthes-phase02.pickle', 'wb') as file:
#     pickle.dump(lengths, file, pickle.HIGHEST_PROTOCOL)

### Loading our lengthes

In [56]:
with open('results/lengthes-phase02.pickle', 'rb') as file:
    loaded_lengthes = pickle.load(file)

In [57]:
lengths = loaded_lengthes

## Making champion list

In [58]:
def make_champion_list(dictionary, r):
    champion_list={}
    
    for term, (total_count, tf) in dictionary.items():
        best_r = dict(Counter(tf).most_common(r))
        champion_list[term] = [total_count, best_r]
    return champion_list

Running this cell below will take:

    CPU times: user 781 ms, sys: 8.02 ms, total: 789 ms
    Wall time: 786 ms

In [59]:
%%time
my_champs = make_champion_list(my_dict, 10)

CPU times: user 585 ms, sys: 15 ms, total: 600 ms
Wall time: 598 ms


## Normalizing query

In [60]:
def normalize_query(query):
    word_frequencies_in_query = preprocess(query)

    for word, word_frequency in word_frequencies_in_query.items():
        word_frequencies_in_query[word] += log_frequency_weighting(word_frequency)
    
    return word_frequencies_in_query

## Setting our K, N

In [61]:
N = len(df)
K = 5

## Cosine similarity

In [62]:
def cosine_score(normalized_query, dictionary):
    similarities = {} 
    for query_term, tf in normalized_query.items():
        if query_term in dictionary.keys():
            idf = math.log10(N / dictionary[query_term][0])

            for docid, weighted_term_frequency in dictionary[query_term][1].items():
                if docid in similarities.keys():
                    similarities[docid] += weighted_term_frequency * tf * idf
                else:
                    similarities[docid] = weighted_term_frequency * tf * idf
                
    for docid in similarities.keys():
        similarities[docid] = similarities[docid] / lengths[docid]
        
    return similarities

In [63]:
import heapq

def best_scores(scores):
    best_scores =[]
    heap = []
    
    for docid, score in scores.items():
        heapq.heappush(heap,(-score, docid))
        
    for i in range(min(K, len(heap))):
        best_scores.append(heapq.heappop(heap))
        
    return best_scores

In [64]:
def query(q, champs=0): 
    normalized_query = normalize_query(q)
    best_r = {}
    
    if champs:
        best_r = cosine_score(normalized_query, my_champs)
        if len(best_r) < K:
            best_r.update(cosine_score(normalized_query, my_dict))
    else:
        best_r = cosine_score(normalized_query, my_dict)
        
    return best_scores(best_r)

In [65]:
def find_docs(top_scores, data):
    results=[]
    
    for t in top_scores:
        docid = t[1]
        url = data[data["id"]==t[1]]["url"].iloc[0]
        content = df[df['id'] == docid]['content'].iloc[0]
        
        results.append((docid, url, content))
        
    return results

In [69]:
def display_results(q):
    top_scores = query(q)
    results = find_docs(top_scores, df)
    
    if len(results) == 0:
        print('Nothing found.')
    else:
        for docid, url, content in results:
            print(' doc id: {}'.format(docid))
            print('    url: {}'.format(url))
            print('content:', end='')
            print(content)

In [70]:
%%time
display_results('کنفدراسیون فوتبال آسیا')

 doc id: 5763
    url: https://www.farsnews.ir/news/14001009000589/AFC-زمان-بررسی-پرونده-حضور-پرسپولیس-در-لیگ-قهرمانان-آسیا-را-مشخص-کرد
content:
به گزارش خبرگزاری فارس و به نقل از سایت باشگاه پرسپولیس، کنفدراسیون فوتبال آسیا در نامه‌ای دریافت جوابیه پرسپولیس را تایید کرد.  این کنفدراسیون اعلام داشت؛ این موضوع در جلسه اعضای کنترل ورودی کنفدراسیون پنج‌شنبه، ششم ژانویه ۲۰۲۲ برابر با ۱۶ دیماه مورد بررسی قرار خواهد گرفت. گفتنی است؛ بنابر اعلام کنفدراسیون فوتبال آسیا، آیین قرعه‌کشی لیگ قهرمانان آسیا ۲۰۲۲، روز دوشنبه ۲۷ دیماه در مقر این کنفدراسیون برگزار می‌شود. انتهای پیام/



 doc id: 3764
    url: https://www.farsnews.ir/news/14001104000657/بازدید-ناظران-AFC-از-استادیوم-آزادی-عکس
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، ناظران کنفدراسیون فوتبال آسیا امروز با حضور در ورزشگاه آزادی از قسمت های مختلف این استادویم دیدن کردند. سهیل مهدی هم ناظران AFC را همراهی کرد.    انتهای پیام/



 doc id: 5071
    url: https://www.farsnews.ir/news/14001019000762/درخواست-پرسپولیس-از-AFC-برای-ارائه-دلای

# Reporting results

## 1.

In [87]:
number_in_all_collection = 4

In [88]:
def find_rare_terms(number_of_occurances):
    for term, (count, tf) in my_dict.items():
        if count < number_in_all_collection:
            print(' term:', term)
            print('count:', count)
            print('   tf:', tf)
            print()

            number_of_occurances -= 1

        if number_of_occurances == 0:
            break

### تحلیل کوئری های آورده شده:
متن هر کوئری به همراه نتایج آورده شده، و با خواندن آن می توان پی برد که کوئری های ما خواسته های کاربر را براورده کرده اند.

### a. simple and frequenty one term query

In [89]:
display_results('کنفدراسیون')

 doc id: 5763
    url: https://www.farsnews.ir/news/14001009000589/AFC-زمان-بررسی-پرونده-حضور-پرسپولیس-در-لیگ-قهرمانان-آسیا-را-مشخص-کرد
content:
به گزارش خبرگزاری فارس و به نقل از سایت باشگاه پرسپولیس، کنفدراسیون فوتبال آسیا در نامه‌ای دریافت جوابیه پرسپولیس را تایید کرد.  این کنفدراسیون اعلام داشت؛ این موضوع در جلسه اعضای کنترل ورودی کنفدراسیون پنج‌شنبه، ششم ژانویه ۲۰۲۲ برابر با ۱۶ دیماه مورد بررسی قرار خواهد گرفت. گفتنی است؛ بنابر اعلام کنفدراسیون فوتبال آسیا، آیین قرعه‌کشی لیگ قهرمانان آسیا ۲۰۲۲، روز دوشنبه ۲۷ دیماه در مقر این کنفدراسیون برگزار می‌شود. انتهای پیام/



 doc id: 4988
    url: https://www.farsnews.ir/news/14001020000675/موافقت-AFC-با-درخواست-پرسپولیس
content:
به گزارش خبرگزاری فارس  و به نقل از سایت باشگاه پرسپولیس، در پی اعلام رای عدم صدور مجوز حرفه‌ای از سوی کنفدراسیون، این باشگاه، دیروز در مدت تعیین شده، درخواست شرح تفصیلی را طی نامه‌ای به AFC، اعلام داشت. کنفدراسیون فوتبال آسیا، دریافت و تایید این درخواست را امروز در نامه‌ای به باشگاه اطلاع داد و اعلام کرد، شرح تفص

### b. simple and frequenty multiple term query

In [90]:
display_results('ورزشگاه آزادی')

 doc id: 3525
    url: https://www.farsnews.ir/news/14001107000594/حضور-بانوان-در-ورزشگاه-آزادی-در-جریان-دیدار-ایران-و-عراق-تصاویر-و-فیلم
content:
به گزارش خبرگزاری فارس، تیم های فوتبال ایران و عراق از مرحله گروهی مقدماتی راه یابی به جام جهانی قطر در ورزشگاه آزادی به مصاف هم رفته اند. در این دیدار مجوز حضور 10 درصد از هواداران را در ورزشگاه داده اند و در این میان هم بانوان هم در ورزشگاه آزادی حضور دارند. بانوان کشورمان پرچم های سه رنگ ایران را در دست داشتند و حضور تعدادی از بانوان محجبه صحنه های جالبی را رقم زد. **تصاویری از حضور بانوان در ورزشگاه آزادی را در زیر ببینید.                 *این فیلم را ببینید.       انتهای پیام/



 doc id: 4761
    url: https://www.farsnews.ir/news/14001023000568/سکوهای-آزادی-سرخ-شد
content:
به گزارش خبرگزاری فارس،کانون هواداران پرسپولیس، مانند دیگر بازی‌های خانگی  ورزشگاه آزادی را مزین به پرچم‌ها و نمادهای پرسپولیسی کرد. گروه داوطلبان کانون هواداران پرسپولیس برای آخرین بازی نیم فصل لیگ بیست و یکم خلیج فارس  سکوهای خالی آزادی را در غیاب هواداران پرشور، س

### c. hard and rare one term query

finding a rare term in dictionary:

In [91]:
find_rare_terms(1)

 term: مرحله‌پلی‌آف‌لیگ
count: 2
   tf: {4: 1.0, 2138: 1.0}



In [92]:
display_results('مرحله‌پلی‌آف‌لیگ 2')

 doc id: 4
    url: https://www.farsnews.ir/news/14001224000947/لیگ‌برتر-بسکتبال|‌-نخستین-پیروزی-شهرداری-گرگان-و-ذوب‌آهن-در-پلی‌آف
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، در نخستین بازی از روز دوم مرحله پلی‌آف لیگ برتر بسکتبال، تیم‌های شهرداری گرگان و اکسون مقابل هم به میدان رفتند که گرگانی‌ها موفق شدند 80 بر 71 به برتری برسند. در این مسابقه پری پتی با کسب 26 امتیاز و 6 ریباند، پرامتیازترین بازیکن گرگان و میدان بود. در دیگر بازی برگزار شده ذوب آهن که با صدرنشینی به پلی‌آف رسیده بود، در دیدار با صنایع هرمزگان 73 بر 70 به برتری رسید. انتهای پیام/    



 doc id: 2138
    url: https://www.farsnews.ir/news/14001125000437/برنامه-بازی‌های-پلی‌آف-لیگ-برتر-والیبال-مشخص-شد
content:
به گزارش خبرگزاری فارس، جلسه قرعه کشی مرحله نخست پلی‌آف مسابقات والیبال قهرمانی باشگاه‌های برتر مردان ایران جام «سردار دل‌ها» ساعت 10 امروز (دوشنبه 25 بهمن‌) به‌صورت آنلاین برگزار شد و برنامه دیدارهای تیم‌های شرکت کننده در مراحل پلی‌آف شامل مراحل یک چهارم‌نهایی، نیمه‌نهایی و نهایی مشخص شد. دیدارهای مرحله یک‌چه

checking a term that is not in dictionary:

In [93]:
print(my_dict.get('آسترازنکا', -1))

-1


In [94]:
display_results('آسترازنکا')

Nothing found.


### d. hard and rare multiple term query

finding some hard and rare multiple terms in dictionary:

In [95]:
find_rare_terms(3)

 term: مرحله‌پلی‌آف‌لیگ
count: 2
   tf: {4: 1.0, 2138: 1.0}

 term: فیدر
count: 2
   tf: {5: 1.0, 132: 1.0}

 term: ‌ندا‌شهسواری
count: 1
   tf: {5: 1.0}



In [96]:
display_results('مرحله‌پلی‌آف‌لیگ فیدر ‌ندا‌شهسواری')

 doc id: 132
    url: https://www.farsnews.ir/news/14001222000537/مسابقات-تنیس-روی-میز-قطر|-شکست-تیم-دو-نفره-بانوان-در-گام-نخست
content:
به گزارش خبرگزاری فارس، در گام نخست مسابقات دونفره فیدر دوحه قطر ، تیم دوبل مهشید اشتری و ندا شهسواری در مصاف با نمایندگان کره جنوبی با حساب ۳ بر صفر شکست خوردند از دور مسابقات کنار رفتند. در این دیدار نمایندگان تنیس روی میز ایران در دو گیم ۱۱ بر ۸ و در گیم سوم ۱۱ بر۶ مغلوب حریف سرسخت کره جنوبی خود شدند. انتهای پیام/



 doc id: 926
    url: https://www.farsnews.ir/news/14001211000910/درخشش-دوباره-بازیکن-36-ساله-اعضای-تیم‌ملی-تنیس-روی-میز-مشخص-شدند
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، رقابت‌های انتخابی تیم ملی تنیس روی میز امروز ادامه پیدا کرد و نیما عالمیان و افشین نوروزی با شکست حریفان خود به عضویت تیم ملی درآمدند. پیش از این امیرحسین هدایی در مرحله اول انتخاب شده بود و نوشاد عالمیان هم از سوی فدراسیون منصوب شد. نکته جالب توجه اینکه نوروزی در حال حاضر با 36 سال سن مسن ترین بازیکن تنیس روی میز ایران محسوب می‌شود و همچنان درخشش دارد. در بخش 

multiple terms that are not in dictionary:

In [97]:
display_results('آسترازنکا')

Nothing found.


In [101]:
display_results('میقولی')

Nothing found.


In [102]:
display_results('آسترازنکا میقولی؟')

Nothing found.


## 2.

### Comparing the b,c results with positional index

## b

### 'ورزشگاه آزادی'

### results for phase 01:

We have 366 document(s) that have 2 words of the user query.
We have 1528 document(s) that have 1 words of the user query.

doc id:  2
title:  محل برگزاری نشست‌های خبری سرخابی‌ها؛ مجیدی در سازمان لیگ، گل‌محمدی در تمرین پرسپولیس
content:

به گزارش خبرگزاری فارس، نشست خبری پیش از مسابقه سرمربیان دو تیم پرسپولیس و استقلال از هفته بیست و سوم لیگ برتر با  مدیریت سازمان لیگ و هماهنگی باشگاه میزبان( پرسپولیس)  به شرح زیر  برگزار می شود: چهارشنبه ۲۵ اسفند ساعت ۱۵ فرهاد مجیدی سرمربی استقلال در سازمان لیگ فوتبال ایران ساعت ۱۳:۳۰ یحیی گل محمدی سرمربی پرسپولیس در ورزشگاه شهید کاظمی مسابقه دو تیم روز پنجشنبه در ورزشگاه آزادی برگزار می شود. انتهای پیام/




doc id:  6
title:  اعلام برنامه نشست خبری گل محمدی/ مجیدی هم باید به محل تمرین پرسپولیس برود!
content:

به گزارش خبرگزاری فارس و به نقل از  سایت باشگاه پرسپولیس، نشست خبری پیش از مسابقه سرمربیان دو تیم پرسپولیس و استقلال از هفته بیست و سوم لیگ برتر (جام خلیج فارس) با  مدیریت سازمان لیگ و هماهنگی باشگاه میزبان (پرسپولیس) در ورزشگاه شهید کاظمی به شرح زیر  برگزار می‌شود: چهارشنبه ۲۵ اسفند ساعت ۱۳ فرهاد مجیدی سرمربی استقلال ساعت ۱۳:۳۰ یحیی گل محمدی سرمربی پرسپولیس  مسابقه دو تیم روز پنجشنبه در ورزشگاه آزادی برگزار می‌شود.  به گزارش خبرگزاری فارس، پیش از این باشگاه استقلال اعلام کرده بود قرار است نشست خبری فرهاد مجیدی، سرمربی این تیم از ساعت ۱۵ در سازمان لیگ برگزار شود. انتهای پیام /




doc id:  10
title:  تراکتور در آستانه بازگشت به خانه اصلی/ نواقص ورزشگاه یادگار امام بررسی شد
content:

به گزارش خبرگزاری فارس و به نقل از سایت باشگاه تراکتور، میر معصوم سهرابی، مدیرعامل این باشگاه به همراه نمایندگانی از استانداری آذربایجان شرقی، معاون عملیات نیروی انتظامی استان، نمایندگانی از یگان ویژه استان، مدیرکل ورزش و جوانان استان، نماینده اداره ورزش و جوانان شهرستان تبریز و مدیر ورزشگاه یادگار امام امروز (سه شنبه ۲۴ اسفند) از بخش‌های مختلف ورزشگاه یادگار امام (ره) تبریز بازدید کرده و آخرین وضعیت و نواقصات این مجموعه برای میزبانی از هواداران تراکتور را بررسی کردند.  طی این بازدید، تمهیدات لازم در راستای آماده‌سازی این ورزشگاه برای میزبانی از بازی‌های تیم تراکتور اندیشیده شد. انتهای پیام/




doc id:  12
title:  گزارش تمرین پرسپولیس| روحیه بالای شاگردان گل محمدی در آستانه دربی
content:

به گزارش خبرگزاری فارس، تمرین امروز تیم فوتبال پرسپولیس از ساعت 11:45 در هوای مطبوع ورزشگاه شهید کاظمی زیر نظر کادر فنی با شادابی و روحیه بالای سرخ پوشان برگزار شد. * بازیکنان ابتدا زیر نظر مربی بدنساز تیم با انجام حرکات کششی، نرمشی و دوهای کوتاه، بدن‌ها را گرم و در اختیار کادر فنی قرار گرفتند. * سپس در قالب سه گروه به تمرین پرداخته و در ادامه نیز مرور کارهای تاکتیکی و فوتبال هدفمند در دستور کار قرار گرفت. * تمرین امروز حدود 50 دقیقه به طول انجامید و کلیه سرخپوشان به جز سیامک نعمتی در برنامه‌های کادر فنی حاضر بودند. در پایان نیز بازیکنان بدن‌ها را سرد و راهی رختکن شدند. تیم پرسپولیس از ساعت 17:30 پنجشنبه در چارچوب دیدارهای هفته بیست و سوم لیگ برتر به مصاف استقلال می رود. انتهای پیام/




doc id:  19
title:  بارسلونا با بازیکن میلان به توافق رسید / 3 مدافع رایگان در لیست ژاوی
content:

به گزارش خبرگزاری فارس، جرارد رومرو، خبرنگار اسپانیایی از توافق «فرانک کسیه» با بارسلونا خبر داد. این بازیکن ساحل عاجی علاقه ندارد با میلان قرارداد خود را تمدید کند و به همین دلیل تصمیم گرفته در پایان فصل به صورت بازیکن آزاد راهی بارسلونا شود.  کسیه در خط میانی روسونری نمایش خوبی در این 2 فصل اخیر داشت. او به یکی از مهره‌های بزرگ پیولی نام گرفته و توانسته در این فصل 29 بازی انجام دهد.  بارسلونا در تابستان قصد دارد برای جذب بازیکنان آزادی مثل کریستینسن و آزپیلیکوئتا (مدافعان چلسی) و نصیر مزروعی (دفاع راست آژاکس) تلاش کند. طبق گزارش‌ها کریستینسن نیز با آبی‌اناری‌ها به توافق رسیده و به احتمال فراوان راهی نیوکمپ خواهد شد. آبی‌اناری‌ها تمام ثروت و توان خود را گذاشته‌اند تا در پنجره نقل و انتقالات تابستانی «ارلینگ هالند» را خریداری کنند. طبق گزارش‌ رسانه‌های اروپایی در چند اخیر منچسترسیتی اصلی‌ترین گزینه انتقال این بازیکن است.  انتهای پیام/

### results for phase 02:

 doc id: 3525
    url: https://www.farsnews.ir/news/14001107000594/حضور-بانوان-در-ورزشگاه-آزادی-در-جریان-دیدار-ایران-و-عراق-تصاویر-و-فیلم
content:
به گزارش خبرگزاری فارس، تیم های فوتبال ایران و عراق از مرحله گروهی مقدماتی راه یابی به جام جهانی قطر در ورزشگاه آزادی به مصاف هم رفته اند. در این دیدار مجوز حضور 10 درصد از هواداران را در ورزشگاه داده اند و در این میان هم بانوان هم در ورزشگاه آزادی حضور دارند. بانوان کشورمان پرچم های سه رنگ ایران را در دست داشتند و حضور تعدادی از بانوان محجبه صحنه های جالبی را رقم زد. **تصاویری از حضور بانوان در ورزشگاه آزادی را در زیر ببینید.                 *این فیلم را ببینید.       انتهای پیام/



 doc id: 4761
    url: https://www.farsnews.ir/news/14001023000568/سکوهای-آزادی-سرخ-شد
content:
به گزارش خبرگزاری فارس،کانون هواداران پرسپولیس، مانند دیگر بازی‌های خانگی  ورزشگاه آزادی را مزین به پرچم‌ها و نمادهای پرسپولیسی کرد. گروه داوطلبان کانون هواداران پرسپولیس برای آخرین بازی نیم فصل لیگ بیست و یکم خلیج فارس  سکوهای خالی آزادی را در غیاب هواداران پرشور، سرخپوش کردند. پرسپولیس ساعت 17:30 امروز در ورزشگاه آزادی به مصاف فجرشهید سپاسی می‌رود. انتهای پیام/



 doc id: 3764
    url: https://www.farsnews.ir/news/14001104000657/بازدید-ناظران-AFC-از-استادیوم-آزادی-عکس
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، ناظران کنفدراسیون فوتبال آسیا امروز با حضور در ورزشگاه آزادی از قسمت های مختلف این استادویم دیدن کردند. سهیل مهدی هم ناظران AFC را همراهی کرد.    انتهای پیام/



 doc id: 411
    url: https://www.farsnews.ir/news/14001218000515/برنامه-مرحله-یک-چهارم-نهایی-جام-حذفی-اعلام-شد-فروردین-40-رقابت
content:
به گزارش خبرگزاری فارس، برنامه مسابقات مرحله یک چهارم نهایی جام حذفی( یادواره آزادسازی خرمشهر) فصل ۱۴۰۱-۱۴۰۰ اعلام شد. برنامه مسابقات به شرح زیر است: شنبه ۲۰ فروردین  مس کرمان – خیبر خرم آباد – ساعت ۲۰ - ورزشگاه شهید باهنر کرمان خلیج فارس ماهشهر – مس رفسنجان – ساعت ۲۰:۱۵ – ورزشگاه متعاقبا اعلام می شود. پرسپولیس – آلومینیوم اراک – ساعت ۲۰:۱۵ – ورزشگاه آزادی تهران دوشنبه ۲۲ فروردین استقلال – نساجی مازندران – ساعت ۲۰:۱۵ – ورزشگاه آزادی تهران انتهای پیام/



 doc id: 5947
    url: https://www.farsnews.ir/news/14001007000217/پخش-زنده-نشست-خبری-فرهاد-مجیدی-پیش-از-دیدار-استقلال-و-فولاد
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، نشست خبری فرهاد مجیدی پیش از دیدار استقلال و فولاد در هفته دوازدهم لیگ برتر در ورزشگاه آرارات برگزار شد. برای تماشای زنده نشست خبری سرمربی استقلال که در حال برگزاری است اینجا را کلیک کنید. دیدار دو تیم استقلال و فولاد ساعت 18 فردا در ورزشگاه آزادی برگزار می شود. انتهای پیام/

## d.

### 'مرحله‌پلی‌آف‌لیگ فیدر ‌ندا‌شهسواری'

### results for phase 01:

We have 5 document(s) that have 1 words of the user query.

doc id:  132
title:  مسابقات تنیس روی میز قطر| شکست تیم دو نفره بانوان در گام نخست
content:

به گزارش خبرگزاری فارس، در گام نخست مسابقات دونفره فیدر دوحه قطر ، تیم دوبل مهشید اشتری و ندا شهسواری در مصاف با نمایندگان کره جنوبی با حساب ۳ بر صفر شکست خوردند از دور مسابقات کنار رفتند. در این دیدار نمایندگان تنیس روی میز ایران در دو گیم ۱۱ بر ۸ و در گیم سوم ۱۱ بر۶ مغلوب حریف سرسخت کره جنوبی خود شدند. انتهای پیام/




doc id:  5
title:  مسابقات تنیس روی میز فیدر قطر| هر4 بانوی ملی پوش ایران حذف شدند
content:

به گزارش خبرگزاری فارس، چهار بانوی تنیس روی میز ایران با باخت در مصاف با رقبای خود در جدول ۶۴ نفره رقابت‌های فیدر دوحه قطر از راهیابی به مرحله بعد بازماندند. ندا شهسواری در مصاف با هوانگ با رنکینگ ۳۸۲ دنیا از چین تایپه با حساب ۴ بر صفر مغلوب شد. شهسواری در گیم اول ۱۱ بر ۷، گیم دوم ۱۱ بر ۲ و در دو گیم متوالی با حساب ۱۱ بر ۶ متوقف و از راهیابی به مرحله بعد بازماند. مهشید اشتری در مصاف با چانگ لی از مالزی با حساب ۴ بر ۲ شکست خورد. حریف اشتری دارنده رنک ۲۳۷ دنیا بود که نماینده کشورمان در دو گیم با حساب ۱۱ بر ۸ شکست خورد و ‌ گیم سوم ۱۱ بر ۷ به سود اشتری خاتمه یافت. گیم چهارم ۱۱ بر ۱۳ به سود حریف به اتمام رسید و اشتری در گیم چهارم ۱۳ بر ۱۱ برابر حریف به برتری دست یافت اما در پایان گیم ششم حریف مالزیایی ۱۲ بر ۱۰ از سد اشتری گذشت. مریم صامت در تقابل با النا از رومانی دارنده رنک ۲۱۵ با نتیجه ۴ بر ۳ شکست خورد. گیم اول ۱۱ بر ۷ به سود حریف به پایان رسید ، صامت در دو گیم متوالی ۱۱ بر ۶ و سپس ۱۱ بر ۷ به پیروزی دست یافت، حریف رومانیایی در گیم چهارم ۱۱ بر ۸ از سد صامت عبور کرد و بازی در گیم پنجم ۱۴ بر ۱۲ به نفع صامت خاتمه یافت. حریف در پایان گیم های ششم و هفتم با برتری ۱۱ بر ۵ از سد نماینده کشورمان عبور کرد. شیما صفایی با قبول شکست ۴ بر صفر مقابل نماینده هند از راهیابی به دور بعد بازماند. ۱۱ بر ۵ در گیم اول ، ۱۳ بر ۱۱ در گیم دوم، ۱۱ بر ۵ و ۱۱ بر ۹ نتایج رقم خورده در این دیدار به نفع حریف هندی صفایی بود. انتهای پیام/




doc id:  3175
title:  سقوط برادران عالمیان در رنکینگ جهانی حاصل غیبت در تورهای بین المللی
content:

به گزارش خبرنگار ورزشی خبرگزاری فارس، در تازه ترین رنکینگ اعلام شده از فدراسیون جهانی تنیس روی میز جدید نوشاد و نیما عالمیان سقوط 4 و 5 پله ای را تجربه کردند. نوشاد عالمیان که در رده 70 قرار داشت به رتبه 74 سقوط کرد و نیما نیز از 127 به 132 رسید. در این بین امیر حسین هدایی و امین احمدیان دیگر لژیونرهای تنیس روی میز صعود داشتند. هدایی از رتبه 185 به 154 رسید و احمدیان نیز با صعود 8 پله‌ای در جایگاه 190 ایستاد. در بخش بانوان نیز ندا شهسواری با 4 پله صعود در رتبه 211 قرار گرفت و همچنان نفر اول ایران است. مهشید اشتری و مریم صامت نیز به ترتیب رده های 342 و 413 را از آن خود کردند. انتهای پیام/




doc id:  978
title:  ملی پوشان تنیس روی میز بانوان مشخص شدند
content:

به گزارش خبرگزاری فارس، در پایان مرحله دوم ماراتن انتخابی تیم ملی بزرگسالان بانوان تنیس روی میز، پریناز حاجیلو از البرز و مهشید اشتری از خراسان رضوی با کسب بیشترین امتیاز به عضویت تیم ملی در آمدند. حاجیلو با کسب ۲۰ امتیاز و اشتری با ۱۷ امتیاز پس از ندا شهسواری از کرمانشاه که توسط کادر فنی انتصاب شد و بعد از شیما صفایی از همدان که در دور اول جواز حضور در تیم ملی را کسب کردند، صاحب لباس تیم ملی شدند. بدین ترتیب برای حضور در بازی های کشورهای اسلامی و بازی های آسیایی ترکیب تیم ملی تنیس روی میز بانوان کشورمان مشخص شد. پنجمین عضو تیم ملی بانوان جهت حضور در رقابت های تیمی قهرمانی جهان با برگزاری مرحله آخر انتخابی مشخص خواهد شد. رده بندی دور دوم انتخابی تیم ملی بانوان: ۱-پریناز حاجیلو ۲۰ امتیاز ۲-مهشید اشتری ۱۷ امتیاز ۳-الینا رحیمی ۱۵ امتیاز ۴-سارا شهسواری ۱۲ امتیاز ۵-کیمیا رستمی ۱۰ امتیاز ۶-فاطمه جمالی فر ۸ امتیاز ۷-فاطمه یاری ۵ امتیاز ۸-مریم فرعی انصراف انتهای پیام/




doc id:  926
title:  درخشش دوباره بازیکن 36 ساله/اعضای تیم‌ملی تنیس روی میز مشخص شدند
content:

به گزارش خبرنگار ورزشی خبرگزاری فارس، رقابت‌های انتخابی تیم ملی تنیس روی میز امروز ادامه پیدا کرد و نیما عالمیان و افشین نوروزی با شکست حریفان خود به عضویت تیم ملی درآمدند. پیش از این امیرحسین هدایی در مرحله اول انتخاب شده بود و نوشاد عالمیان هم از سوی فدراسیون منصوب شد. نکته جالب توجه اینکه نوروزی در حال حاضر با 36 سال سن مسن ترین بازیکن تنیس روی میز ایران محسوب می‌شود و همچنان درخشش دارد. در بخش بانوان نیز شیما صفایی، پریناز حاجیلو و مهشید اشتری انتخاب شدند و ندا شهسواری نیز از سوی فدراسیون انتصاب شد. انتهای پیام/  

### results for phase 02:

doc id: 132
    url: https://www.farsnews.ir/news/14001222000537/مسابقات-تنیس-روی-میز-قطر|-شکست-تیم-دو-نفره-بانوان-در-گام-نخست
content:
به گزارش خبرگزاری فارس، در گام نخست مسابقات دونفره فیدر دوحه قطر ، تیم دوبل مهشید اشتری و ندا شهسواری در مصاف با نمایندگان کره جنوبی با حساب ۳ بر صفر شکست خوردند از دور مسابقات کنار رفتند. در این دیدار نمایندگان تنیس روی میز ایران در دو گیم ۱۱ بر ۸ و در گیم سوم ۱۱ بر۶ مغلوب حریف سرسخت کره جنوبی خود شدند. انتهای پیام/



 doc id: 926
    url: https://www.farsnews.ir/news/14001211000910/درخشش-دوباره-بازیکن-36-ساله-اعضای-تیم‌ملی-تنیس-روی-میز-مشخص-شدند
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، رقابت‌های انتخابی تیم ملی تنیس روی میز امروز ادامه پیدا کرد و نیما عالمیان و افشین نوروزی با شکست حریفان خود به عضویت تیم ملی درآمدند. پیش از این امیرحسین هدایی در مرحله اول انتخاب شده بود و نوشاد عالمیان هم از سوی فدراسیون منصوب شد. نکته جالب توجه اینکه نوروزی در حال حاضر با 36 سال سن مسن ترین بازیکن تنیس روی میز ایران محسوب می‌شود و همچنان درخشش دارد. در بخش بانوان نیز شیما صفایی، پریناز حاجیلو و مهشید اشتری انتخاب شدند و ندا شهسواری نیز از سوی فدراسیون انتصاب شد. انتهای پیام/  



 doc id: 3175
    url: https://www.farsnews.ir/news/14001112000231/سقوط-برادران-عالمیان-در-رنکینگ-جهانی-حاصل-غیبت-در-تورهای-بین-المللی
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، در تازه ترین رنکینگ اعلام شده از فدراسیون جهانی تنیس روی میز جدید نوشاد و نیما عالمیان سقوط 4 و 5 پله ای را تجربه کردند. نوشاد عالمیان که در رده 70 قرار داشت به رتبه 74 سقوط کرد و نیما نیز از 127 به 132 رسید. در این بین امیر حسین هدایی و امین احمدیان دیگر لژیونرهای تنیس روی میز صعود داشتند. هدایی از رتبه 185 به 154 رسید و احمدیان نیز با صعود 8 پله‌ای در جایگاه 190 ایستاد. در بخش بانوان نیز ندا شهسواری با 4 پله صعود در رتبه 211 قرار گرفت و همچنان نفر اول ایران است. مهشید اشتری و مریم صامت نیز به ترتیب رده های 342 و 413 را از آن خود کردند. انتهای پیام/



 doc id: 978
    url: https://www.farsnews.ir/news/14001210000627/ملی-پوشان-تنیس-روی-میز-بانوان-مشخص-شدند
content:
به گزارش خبرگزاری فارس، در پایان مرحله دوم ماراتن انتخابی تیم ملی بزرگسالان بانوان تنیس روی میز، پریناز حاجیلو از البرز و مهشید اشتری از خراسان رضوی با کسب بیشترین امتیاز به عضویت تیم ملی در آمدند. حاجیلو با کسب ۲۰ امتیاز و اشتری با ۱۷ امتیاز پس از ندا شهسواری از کرمانشاه که توسط کادر فنی انتصاب شد و بعد از شیما صفایی از همدان که در دور اول جواز حضور در تیم ملی را کسب کردند، صاحب لباس تیم ملی شدند. بدین ترتیب برای حضور در بازی های کشورهای اسلامی و بازی های آسیایی ترکیب تیم ملی تنیس روی میز بانوان کشورمان مشخص شد. پنجمین عضو تیم ملی بانوان جهت حضور در رقابت های تیمی قهرمانی جهان با برگزاری مرحله آخر انتخابی مشخص خواهد شد. رده بندی دور دوم انتخابی تیم ملی بانوان: ۱-پریناز حاجیلو ۲۰ امتیاز ۲-مهشید اشتری ۱۷ امتیاز ۳-الینا رحیمی ۱۵ امتیاز ۴-سارا شهسواری ۱۲ امتیاز ۵-کیمیا رستمی ۱۰ امتیاز ۶-فاطمه جمالی فر ۸ امتیاز ۷-فاطمه یاری ۵ امتیاز ۸-مریم فرعی انصراف انتهای پیام/



 doc id: 5213
    url: https://www.farsnews.ir/news/14001018000410/۹-غایب-در-جلسه-کمیسیون-ورزشکاران-برنامه‌ریزی-برای-گرفتن-بودجه-IOC
content:
به گزارش خبرنگار ورزشی خبرگزاری فارس، سومین نشست کمیسیون ورزشکاران امروز شنبه در کمیته ملی المپیک برگزار شد. در نشست امروز، هادی ساعی، سجاد گنج‌زاده، امین بوداغی، محمد بیداریان، بهداد سلیمی، ثریا آقایی، حمیده عباسعلی، مجتبی عابدینی، شهربانو منصوریان، زهرا نعمتی و مصطفی نودهی در نشست امروز حاضر بودند. حسین ساوه‌شمشکی، حسن یزدانی، محمدرضا گرایی، احسان حدادی، فرزانه فصیحی، ندا شهسواری، نجمه خدمتی، ذبیح‌الله پورشیب، محمد باقری معتمد در نشست حضور نداشتند. محمد بیداریان سخنگوی کمیسیون ورزشکاران درباره مصوبات امروز، گفت: طبق برنامه دعوت نامه به قهرمانان ارسال شده بود و ۱۱ نفر حضور داشتند. در خصوص صدور کارت کمیته صحبت و قرار شد فدراسیون‌ها همکاری لازم را داشته باشد. بیداریان همچنین در خصوص برنامه‌های جلسه بعدی، عنوان کرد: از کارشناس بیمه دعوت می‌شود. کارگروهی تشکیل خواهد شد تا در برنامه‌های کمیته از این عزیزان استفاده شود. در خصوص پرداخت بودجه از سوی IOC برنامه‌های خود را طراحی و تدوین خواهیم کرد و قرار است هماهنگی‌ها انجام شود. نامه‌نگاری می‌کنیم تا کمیسیون ردیف بودجه مجزا داشته باشد تا در خصوص کمک به ورزشکاران حمایت لازم انجام شود. درخواست برای عضویت یک ورزشکار از کمیسیون در صندوق حمایت از ورزشکاران از برنامه‌های دیگر ماست. وی درباره اینکه قرار بود امروز وزیر در جلسه شما شرکت کند اما این اتفاق رخ نداد، عنوان کرد: وزارت درگیر انتخابات چند فدراسیون است، قرار شد در شرایط مناسبی از وزیر دعوت شود. وی همچنین در خصوص برگزاری جلسه با ۱۱ قهرمان، تصریح کرد: امروز حضور هادی ساعی پیام خوبی برای همه اعضای کمیسیون داشت، اینکه به تازگی رئیس فدراسیون انتخاب شده جایگاه کمیسیون آنقدر بالا بود که او حضور داشت. من انتخاب او را هم تبریک می‌گویم و امیدوارم جلسات بعدی با حضور پرتعداد برگزار شود. بیداریان در مورد اینکه چند ورزشکار ۳ جلسه غایب بوده‌اند و آیا آنها از کمیسیون خارج می‌شوند، گفت: غیبت چند ورزشکار موجه بود. احسان حدادی در کیش حضور دارد، فصیحی در خارج از کشور است و نجمه خدمتی هم تهران نبود. سخنگوی کمیسیون ورزشکاران درباره غیبت بیش از ۳ جلسه کشتی‌گیران، خاطرنشان کرد: امیدوارم این همدلی به وجود بیاید تا کشتی‌گیران هم در جلسات حضور داشته باشند. انتهای پیام/